src/native/libs/System.Native/pal_time.c

```diff
@@ -83,17 +81,12 @@ int32_t SystemNative_FUTimens(intptr_t fd, TimeSpec* times)
 
 uint64_t SystemNative_GetTimestamp(void)
 {
-#if HAVE_CLOCK_GETTIME_NSEC_NP
-    return clock_gettime_nsec_np(CLOCK_UPTIME_RAW);
-#else
-    struct timespec ts;
-
-    int result = clock_gettime(CLOCK_MONOTONIC, &ts);
-    assert(result == 0); // only possible errors are if MONOTONIC isn't supported or &ts is an invalid address
-    (void)result; // suppress unused parameter warning in release builds
+    return (uint64_t)minipal_hires_ticks();
```

Comment:
You can change the type of PInvoke to int64_t to avoid back and forth casts. The one place where this is PInvoked casts the result back to signed int64_t.